Source/KakapoDB.swift

```diff
@@ -31,6 +31,39 @@ private final class ArrayBox<T> {
     private var value: [T]
 }
 
+private extension Array {
+    
+    func indexOf<T: Storable>(element: T) -> Int? {
+        for (index, object) in enumerate() {
+            if let object = object as? Storable
+                where object.dynamicType == element.dynamicType && object.id == element.id {
+                
+                return index
+            }
+        }
+        
+        return nil
+    }
+    
+    func dataIndexOf<T: Storable>(element: T) -> Int? {
+        let elementData = toData(element.serialize())
+        
+        for (index, object) in enumerate() {
+            if let object = object as? Storable,
+                   elementData = elementData,
+                   objectData = toData(object.serialize())
+                where object.dynamicType == element.dynamicType &&
+                      object.id == element.id &&
+                      objectData.isEqualToData(elementData) {
+                
+                return index
+            }
+        }
+        
+        return nil
+    }
```

Comment:
@alm not quite happy with this two helpers in the extension, suggestion? Can't make the extension constrain the Element to be Storable and thus those ugly casts :(